src/hooks/ContentType.ts

```diff
@@ -3,10 +3,9 @@ import { loadURLContentType } from 'utils/http/loadURLContentType'
 
 export function useContentType(link?: string) {
   const [contentType, setContentType] = useState<string>()
-
   useEffect(() => {
     loadURLContentType(link).then(type => setContentType(type))
   }, [link])
 
-  return contentType
+  return { data: contentType, loading: contentType !== undefined }
```

Comment:
**suggestion** i still don't think `undefined` is a sufficient check for the loading state. What if `type` returned and was undefined? Or if there was an [exception](https://github.com/jbx-protocol/juice-interface/blob/99f6b5fe76cf6233325eb7d80fcc5fb1ae740b31/src/utils/http/loadURLContentType.ts#L9)?

Would suggest we have another state var `loading`, and set it to `false` when `loadURLContentType` finishes loading (success or error)

Reply:
@johnnyd-eth in fact to make our lives way easier, why don't we just implement `useQuery` here? Basically just like https://github.com/jbx-protocol/juice-interface/blob/982ea95c8a8d8302a4b2be60ab8846253053b8e0/src/hooks/ProjectMetadata.ts#L18

Reply:
Yeah good call, much cleaner. Done that